src/System.Runtime/src/System/HashCode.cs

```diff
@@ -313,94 +317,112 @@ public void Add<T>(T value, IEqualityComparer<T> comparer)
+            // anything yet. xxHash32 treats this differently.
 
-                // If the length is less than 3, _v1 -> _v4 don't contain
-                // anything yet. xxHash32 treats this differently.
+            uint hash = (_length ^ position) == 0
+                ? MixEmptyState()
+                : MixState(_v1, _v2, _v3, _v4);
 
-                var hash = (_length ^ position) == 0
-                    ? MixEmptyState()
-                    : MixState(_v1, _v2, _v3, _v4);
+            // Multiply by 4 because we've been counting in ints, not
+            // bytes.
 
-                hash += _length * 4;
+            hash += _length * 4;
 
-                // Mix what remains in the queue
+            // Mix what remains in the queue
 
-                switch (position)
+            if (position > 0)
+            {
+                QueueRound(ref hash, _queue1);
+                if (position > 1)
                 {
-                    case 1:
-                        hash = QueueRound(hash, _queue1);
-                        break;
-
-                    case 2:
-                        hash = QueueRound(hash, _queue1);
-                        hash = QueueRound(hash, _queue2);
-                        break;
-
-                    case 3:
-                        hash = QueueRound(hash, _queue1);
-                        hash = QueueRound(hash, _queue2);
-                        hash = QueueRound(hash, _queue3);
-                        break;
+                    QueueRound(ref hash, _queue2);
+                    if (position > 2)
+                        QueueRound(ref hash, _queue3);
                 }
+            }
 
-                // Final mix
-
-                hash = MixFinal(hash);
+            MixFinal(ref hash);
 
-                return (int)hash;
-            }
+            return (int)hash;
         }
 
-#pragma warning disable 0809
+#       pragma warning disable 0809
+        // Obsolete member 'memberA' overrides non-obsolete member 'memberB'. 
+        // Disallowing GetHashCode is by design
+
+        // * We decided to not override GetHashCode() to produce the hash code 
+        //   as this would be weird, both naming-wise as well as from a behavioral 
+        //   standpoint (GetHashCode() should return the object's hash code, not 
+        //   the one being computed).
+
+        // * Even though ToHashCode() can be called safely multiple times on this
+        //   implementation, it is not part of the contract. If the implementation
+        //   has to change in the future we don't want to worry about people who
+        //   might have incorrectly used this type.
 
         [Obsolete("Use ToHashCode to retrieve the computed hash code.", error: true)]
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public override int GetHashCode() => throw new NotImplementedException();
-
-#pragma warning restore 0809
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Allow this method to show up in the stack trace, self documenting the problem.

Reply:
Worth a commenT